src/build_helper/run.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2012-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Run the command in a background thread. Redirect the output (and error
+//! output) to the screen.
+
+use std::process::{Stdio, Command};
+use std::sync::mpsc::{channel, Receiver};
+use std::thread;
+use std::io::{self, Read, BufRead, BufReader, Write};
+
+/// Run the command in another thread. Print output to screen.
+/// Panics if command failed to run.
+pub trait Run {
+    fn run(&mut self);
+}
```

Comment:
This may be better expressed as just a function in this module as this trait isn't ever implemented for something other than `Command`?